src/Microsoft.OData.Core/Microsoft.OData.Core.csproj

```diff
@@ -95,13 +91,14 @@
     </PackageReference>
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+  <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">

```

Comment:
You should confirm whether a higher version of Microsoft.Extensions.ObjectPool is available for net8.0. Or if it's even required at all

Reply:
Since I was focusing on just making the changes necessary to get this to work. I avoided removing packages, to avoid accidental breakages or lengthening the scope of this PR.

Reply:
@habbes Especially in the case of Microsoft.Bcl.AsyncInterfaces that I flagged elsewhere, my concern is that if it's unneeded and we fail to remove it now, we might not have revisit it again to clean up

Reply:
Removing `Microsoft.Extensions.ObjectPool` reference causes a compilations error. It's not built-in to .NET. It's documentation is not listed as part of .NET but rather in .NET Platform Extensions: https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.objectpool?view=dotnet-plat-ext-8.0

Reply:
However, removing the `Bcl.AsyncInterfaces` references seems to cause no issues.